### PR TITLE
quant docs: add and clean up LayerNorm

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40377 quant docs: add and clean up ELU
* #40346 quant docs: add and clean up BatchNorm{n}d
* #40345 quant docs: add and clean up InstanceNorm{n}d
* #40343 quant docs: add and clean up GroupNorm
* **#40342 quant docs: add and clean up LayerNorm**
* #40341 quant docs: add and clean up hardtanh
* #40340 quant docs: add and clean up hardsigmoid
* #40323 quant docs: clean up hardswish

Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

### CODE REVIEW

**Review by supriyar — APPROVED** [review]

File: torch/nn/qat/modules/normalization.py [hunk 1]
```diff
@@ -182,10 +182,10 @@ def from_float(cls, mod, qconfig=None):
 
 class LayerNorm(nn.LayerNorm):
     r"""
-    A LayerNorm module attached with FakeQuantize modules for both output
-    activation and weight, used for quantization aware training.
+    A LayerNorm module attached with FakeQuantize modules for output
+    activation, used for quantization aware training.
 
-    Similar to `torch.nn.LayerNorm`, with FakeQuantize modules initialized to
+    Similar to :class:`torch.nn.LayerNorm`, with FakeQuantize modules initialized to
```
Comment (raghuramank100) [inline torch/nn/qat/modules/normalization.py:188]: I think we do not need this class. By default, we insert an activation post process after the operation, without needing a new class.

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Lets remove the QAT layer norm class.

**Review by raghuramank100 — APPROVED** [review]
